lib/support/compile-config.js

```diff
@@ -150,17 +150,20 @@ import rule_${ idx } from '${ pkg }/rules/${ ruleName }';
 cache['${ originalPkg }/${ ruleName }'] = rule_${ idx };`;
 }
 
-function computeRuleImport(resolver, pkg, rule) {
+function computeRuleImport(resolver, pkg, originalPkg, rule) {
 
   // local reference, resolved relative to pkg location
   if (rule.startsWith('.')) {
-    const pkgMain = resolver.require.resolve(pkg);
+    const pkgJsonPath = path.dirname(resolver.require.resolve(`${ pkg }/package.json`));
 
-    const pkgDir = path.dirname(pkgMain.split(path.sep).join(path.posix.sep));
+    const pkgMainPath = path.dirname(resolver.require.resolve(pkg));
 
-    return path.posix.normalize(path.posix.join(pkgDir, rule));
-  }
+    const relativePkgMainPath = path.relative(pkgJsonPath, pkgMainPath);
+
+    return path.posix.normalize(`${ originalPkg }/${ relativePkgMainPath }/${ rule }`);
```

Comment:
@philippfromme Here we have to make sure that `relativePkgMainPath` is posix, i.e. pipe it through `win.split(path.sep).join(path.posix.sep)`.

Reply:
The problem is that both `pkgJsonPath` and `pkgMainPath` will be `.` after using `dirname`. So the relative path is wrong.

Reply:
This is what I get on Windows using `posix` explicitly:

```
pkgJsonPath: .
pkgMainPath: .
path before normalizing: bpmnlint-plugin-foreign//./rules/exported-path
```

Reply:
But you're correct, this would cause us to move the special OS treatment within the library code. Let's not do it. My appologies for https://github.com/bpmn-io/bpmnlint/pull/114#discussion_r1270570682. I'll revert to your initial approach and we'll rely on cross platform CI testing to catch this issue.